Feat: 서버 구축

khoon9 opened the pull request:
### Review 유의사항
---
- 채팅 스키마, crud, api 구축했습니다.
- 서버 내부에서 랭체인이 동작할 수 있도록 api와 랭체인을 연동할 수 있는 공간을 마련했습니다.
- 랭체인 로직을 서버 내부에 마련된 공간에 이동시키는 작업이 필요합니다.

khoon9 committed:
Init: Server FastAPI Setting

- FastAPI 기본 구조 세팅
- 데이터베이스 스키마 마이그레이션 설정(by Alembic)

seochan99 commented:
고생하셨어요.

혹 아래 제꺼 리뷰 및 머지 해주실 수 있을까요

seochan99 commented on server/README.md:
친절하고 좋습니다~

seochan99 commented on server/app/database.py:
db url은 뭘로해뒀어용? env파일 공유바랍니닷

seochan99 reviewed (approved):
고생하셨습니다!

khoon9 committed:
Feat: 챗봇 스키마, crud, API 구성

chatRoom:
- name 받아서 chatroom 생성 및 chatroom 식별용 id 반환
chatRecord:
 - is_chatbot을 통해 question(is_chatbot=False)과 response(chatbot=True)를 구분

khoon9 committed:
Chore: 개발 편의 설정

- 매 실행마다 db drop 후 create 되도록 함
- 추후 drop 주석처리 필요

khoon9 committed:
Feat: 서버 API와 랭체인 연동

server/app 내부 부가설명
- routers/chat.py 에서 채팅 REST API 관리
-> crud.py 에서 crud 수행
-> service/chatbot/chat.py 에서 랭체인 주요 동작 수행
